### PR TITLE
Work on code generation for C# / .NET (MAUI)

### PR DESCRIPTION
**Pull request by hansmbakker**
*Issue #, if available:* #977

*Description of changes:*

Export tokens to C# class format, so that they are also usable in C#, and so that you can generate compilable code. This is intended to be usable in .NET MAUI.

- [x] Support for `Color`
- [ ] Support for `number`s / `dimension`s
- [ ] Support for `string`
- [ ] Support for token references
- [ ] Support for grouping tokens in classes, and cross-referencing them
- [ ] Support for effects like `shadow`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by hansmbakker** [comment]
@jorenbroekema

What I am struggling with, is
- style dictionary flattens everything
- how can I have groups of (color) tokens that are exported to different classes, that cross-reference each other?

The use case is this:
I want to have 3 `class`es of `color` tokens:
- ColorTokens: `color tokens.static.text-on-bg.standard.moderate` references `color aliases.neutral.800`
- ColorAliases: `color aliases.neutral.800` references `primitives.grey.800`
- ColorPrimitives: `primitives.grey.800` resolves to `#525252ff`

File: lib/common/templates/dotnet/class.cs.template.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
+/**
+ * @typedef {import('../../../../types/DesignToken.d.ts').TransformedToken} TransformedToken
+ * @typedef {import('../../../../types/Config.d.ts').Config} Config
+ * @typedef {import('../../../../types/File.d.ts').File} File
+ * @typedef {import('../../../../types/Config.d.ts').LocalOptions} LocalOptions
+ */
+
```
Comment (jorenbroekema) [inline lib/common/templates/dotnet/class.cs.template.js:7]: This looks great!

File: package.json [hunk 2]
```diff
@@ -157,5 +157,13 @@
     "unist-util-visit": "^5.0.0",
     "uuid": "^9.0.1",
     "yaml": "^2.3.4"
+  },
+  "mocha": {
```
Comment (jorenbroekema) [inline package.json:161]: Ah cool, then we can also remove these params in the `scripts['test:node']` for brevity right?

**Comment by jorenbroekema** [comment]
It sounds like what you're looking for is something that is similar to [`outputReferences`](https://style-dictionary-v4.netlify.app/reference/hooks/formats/#references-in-output-files), this feature is theoretically implementable in any format but it's not super trivial to do it. I see that you're reusing createPropertyFormatter which has an implementation for outputReferences which is great, that definitely helps a lot, is there something that you feel is missing from that?

File: lib/common/formatHelpers/setCSharpFileProperties.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+/**
+ * Outputs an object with C# format configurations. Sets access modifier.
+ * @memberof module:formatHelpers
+ * @name setCSharpFileProperties
+ * @param {{accessModifier?: string;}} options - The options object declared at configuration
+ * @returns {Object}
+ */
+export default function setCSharpFileProperties(options) {
+  if (typeof options.accessModifier === 'undefined') {
+    options.accessModifier = 'public ';
```
Comment (jorenbroekema) [inline lib/common/formatHelpers/setCSharpFileProperties.js:27]:
```suggestion
    options.accessModifier = 'public';
```

File: lib/common/formatHelpers/setCSharpFileProperties.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  } else {
+    if (options.accessModifier !== '') {
+      options.accessModifier = `${options.accessModifier} `;
```
Comment (jorenbroekema) [inline lib/common/formatHelpers/setCSharpFileProperties.js:30]:
```suggestion
      options.accessModifier = `${options.accessModifier}`;
```

**Comment by jorenbroekema** [comment]
I reviewed your PR thusfar and it looks great! let me know if there's anything I can help with, [you can also reach me on Slack](https://join.slack.com/t/tokens-studio/shared_invite/zt-1p8ea3m6t-C163oJcN9g3~YZTKRgo2hg) if you wanna have a call at some point to flesh out some of the details

**Comment by Kallekro** [comment]
This PR was closed with no comment. I assume the initiative has been dropped?

**Comment by jorenbroekema** [comment]
yeah the PR was abandoned but I'd welcome anyone to continue it!

**Comment by jcfranco** [comment]
@jorenbroekema I’d be interested in helping out here. Is this PR still a good starting point? Any pointers would be greatly appreciated!

**Comment by jorenbroekema** [comment]
Nice! Hard to say to be honest, it's quite a while ago so the merge conflicts might not be worth it. I generally just go file by file and manually cherrypick the changes I need from the outdated branch, in a new branch that's based on latest main. Seems like I liked most of the changes that were done so probably still a good starting point, dont know how much the target platform (C# etc) changed in the meantime.

**Pull request closed without merging**